### PR TITLE
[scripts] Fix vcpkg_fixup_cmake on non Windows platforms

### PR DESCRIPTION
**Pull request by tarcila**
Script was only handling tools executables ending with .exe.
Changed it so anything under /bin/ in transformed.

This fixes for instance FlatcTargets-release.cmake from flatbuffers
port on osx.

### CODE REVIEW

**Comment by Neumann-A** [comment]
Have you tested this on Windows? The bin folder contains the *.dlls and *.pdbs on Windows not only the exe. I tried something with CMAKE_EXECUTABLE_SUFFIX within a port a short while ago on windows but it failed du to CMAKE_EXECUTABLE_SUFFIX not being set to .exe for some reason

**Comment by tarcila** [comment]
I did not test this under Windows for now, as I will only be able to do this later today.

But, I would hope it works as this commit is only about changing references to files inside some
cmake description files, not dealing with the files themselves. And as far as I know those cmake description files only address importable targets -- executables and import libraries (.lib) on Windows, and import lib are not in the bin/ folder.

I'll update the PR when I have tested it

**Comment by Neumann-A** [comment]
> references to files inside some
cmake description files

If those references are pointing to the wrong direction cmake has a problem. 

From the VTK Portfile:
```
if(VTK_WITH_ALL_MODULES)
    file(READ ${CURRENT_PACKAGES_DIR}/share/vtk/VTKTargets-release.cmake VTK_TARGETS_RELEASE_CONTENT)
    string(REPLACE "lib/../XdmfCore.lib" "lib/XdmfCore.lib" VTK_TARGETS_RELEASE_CONTENT "${VTK_TARGETS_RELEASE_CONTENT}")
    string(REPLACE "bin/../XdmfCore.dll" "bin/XdmfCore.dll" VTK_TARGETS_RELEASE_CONTENT "${VTK_TARGETS_RELEASE_CONTENT}")
    string(REPLACE "lib/../vtkxdmf3.lib" "lib/vtkxdmf3.lib" VTK_TARGETS_RELEASE_CONTENT "${VTK_TARGETS_RELEASE_CONTENT}")
    string(REPLACE "bin/../vtkxdmf3.dll" "bin/vtkxdmf3.dll" VTK_TARGETS_RELEASE_CONTENT "${VTK_TARGETS_RELEASE_CONTENT}")
    file(WRITE ${CURRENT_PACKAGES_DIR}/share/vtk/VTKTargets-release.cmake "${VTK_TARGETS_RELEASE_CONTENT}")

    file(READ ${CURRENT_PACKAGES_DIR}/share/vtk/VTKTargets-debug.cmake VTK_TARGETS_DEBUG_CONTENT)
    string(REPLACE "lib/../XdmfCore.lib" "lib/XdmfCore.lib" VTK_TARGETS_DEBUG_CONTENT "${VTK_TARGETS_DEBUG_CONTENT}")
    string(REPLACE "bin/../XdmfCore.dll" "bin/XdmfCore.dll" VTK_TARGETS_DEBUG_CONTENT "${VTK_TARGETS_DEBUG_CONTENT}")
    string(REPLACE "lib/../vtkxdmf3.lib" "lib/vtkxdmf3.lib" VTK_TARGETS_DEBUG_CONTENT "${VTK_TARGETS_DEBUG_CONTENT}")
    string(REPLACE "bin/../vtkxdmf3.dll" "bin/vtkxdmf3.dll" VTK_TARGETS_DEBUG_CONTENT "${VTK_TARGETS_DEBUG_CONTENT}")
    file(WRITE ${CURRENT_PACKAGES_DIR}/share/vtk/VTKTargets-debug.cmake "${VTK_TARGETS_DEBUG_CONTENT}")
endif()
```

So PCL will probably fail with your commit or some external project with wants to install the vtk dlls will fail to do so.

**Comment by tarcila** [comment]
I confirm that trying to solve the problem with executables on non Windows platforms, my commit does break Windows DLL support in the aforementioned files. I'll submit an updated PR later today.

**Comment by tarcila** [comment]
I have pushed an updated version where:
- only .exe are transformed on Windows platforms (previous behaviour)
- all files are transformed on other platforms

As of now, I have been able to successfully build pcl on Windows, vtk, harfbuzz on Windows and OSX, resolving regressions introduced by earlier version of the PR.
Now flatbuffers's flatc is correctly referenced by FlatcTargets-release.cmake on OSX,. It also fixes my in progress Capnproto package.

File: scripts/cmake/vcpkg_fixup_cmake_targets.cmake [hunk 1]
```diff
@@ -27,6 +28,13 @@ function(vcpkg_fixup_cmake_targets)
         set(_vfct_TARGET_PATH share/${PORT})
     endif()
 
+
+    if(NOT VCPKG_CMAKE_SYSTEM_NAME OR VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Windows")
```
Comment (Neumann-A) [inline scripts/cmake/vcpkg_fixup_cmake_targets.cmake]: does this include WindowsStore?

Comment (tarcila) [reply scripts/cmake/vcpkg_fixup_cmake_targets.cmake]: It does not include WindowsStore as I have made a typo: Windows is already tested by the empty VCPKG_CMAKE_SYSTEM_NAME. The second one was intended to be WindowsStore.
Will fix it.

**Comment by ras0219-msft** [comment]
LGTM

**Comment by Rastaban** [comment]
At preliminary glance at the test results tells me they are probably all unrelated failures (flaky or conflicting ports)  I am looking into the failures and will report back anything that looks like it could be valid.

File: ports/grpc/CONTROL [hunk 2]
```diff
@@ -1,4 +1,4 @@
 Source: grpc
-Version: 1.21.1
+Version: 1.21.1-1
```
Comment (Neumann-A) [inline ports/grpc/CONTROL]: If you are already touching grpc you probably also want to move the extra handling of grpc in the toolchain file vcpkg.cmake in the find_package function into a separate vcpkg_cmake_wrapper.cmake.

Comment (Neumann-A) [reply ports/grpc/CONTROL]: ah already merged nevermind
